webapi/src/main/scala/org/knora/webapi/responders/admin/ProjectsResponderADM.scala

```diff
@@ -762,46 +761,48 @@ class ProjectsResponderADM(responderData: ResponderData) extends Responder(respo
             }
 
             // check if the supplied shortname is unique
-            shortnameExists <- projectByShortnameExists(createRequest.shortname)
+            shortnameExists <- projectByShortnameExists(createProjectRequest.shortname)
             _ = if (shortnameExists) {
-                throw DuplicateValueException(s"Project with the shortname: '${createRequest.shortname}' already exists")
+                throw DuplicateValueException(s"Project with the shortname: '${createProjectRequest.shortname}' already exists")
             }
 
             validatedShortcode = StringFormatter.getGeneralInstance.validateProjectShortcode(
-                createRequest.shortcode,
-                errorFun = throw BadRequestException(s"The supplied short code: '${createRequest.shortcode}' is not valid.")
+                createProjectRequest.shortcode,
+                errorFun = throw BadRequestException(s"The supplied short code: '${createProjectRequest.shortcode}' is not valid.")
             )
 
             // check if the optionally supplied shortcode is valid and unique
             shortcodeExists <- {
                 projectByShortcodeExists(validatedShortcode)
             }
             _ = if (shortcodeExists) {
-                throw DuplicateValueException(s"Project with the shortcode: '${createRequest.shortcode}' already exists")
+                throw DuplicateValueException(s"Project with the shortcode: '${createProjectRequest.shortcode}' already exists")
             }
 
-            newProjectIRI = stringFormatter.makeRandomProjectIri(validatedShortcode)
-
+            newProjectIRI: IRI = createProjectRequest.projectIri match {
```

Comment:
@subotic isn't it better to leave this here, because here it runs while holding a lock on the IRI?

Reply:
In the route, i.e., the case class you check if the custom project IRI is valid. Here you would need to additionally check if the custom project IRI is not used already used. At this point, you only now that it is valid, but it doesn't mean that it is unique. This would also mean, that you need a test to check if you get the correct error if you supply an already existing custom project IRI.

Reply:
You are absolutely right, thanks for the heads up. I added validation to route, and checks for uniqueness to the responder + tests for uniqueness.